scripts/configure-pipelines.sh

```diff
@@ -0,0 +1,318 @@
+        --from-literal="webhook-github-secret=${WEBHOOK_SECRET}" \
+        --from-literal="webhook-url=$(kubectl get routes -n "${PIPELINES_NAMESPACE}" pipelines-as-code-controller -o jsonpath="https://{.spec.host}")" >/dev/null
+else
+    WEBHOOK_SECRET="$(kubectl -n "${NAMESPACE}" get secret "${PIPELINES_SECRET_NAME}" ) -o jsonpath="{.data.webhook-github-secret}" | base64 -d"
+fi
+if [ $? -ne 0 ]; then
+    echo "FAIL"
+    exit 1
+fi
+
+if [ "$(kubectl get secret -n "${PIPELINES_NAMESPACE}" "pipelines-as-code-secret" -o name --ignore-not-found | wc -l | tr -d '[:space:]')" = "0" ]; then
+    kubectl -n "${PIPELINES_NAMESPACE}" create secret generic pipelines-as-code-secret \
+        --from-literal github-application-id="${GITHUB__APP__ID}" \
+        --from-literal github-private-key="${GITHUB__APP__PRIVATE_KEY}" \
+        --from-literal webhook.secret="${WEBHOOK_SECRET}" \
+        --dry-run=client -o yaml | kubectl apply -f - >/dev/null
+    
+    if [ $? -ne 0 ]; then
+        echo "FAIL"
+        exit 1
+    fi
+fi
+echo "OK"
+
+# Configure Namespaces
+# Configuring namespaces with needed resources
+echo -n "* Configuring Namespaces: "
+while ! kubectl get secrets -n openshift-pipelines signing-secrets >/dev/null 2>&1; do
+    echo -n "_"
+    sleep 2
+done
+echo -n "."
+COSIGN_SIGNING_PUBLIC_KEY=""
+while [ -z "${COSIGN_SIGNING_PUBLIC_KEY:-}" ]; do
+    echo -n "_"
+    sleep 2
+    COSIGN_SIGNING_PUBLIC_KEY=$(kubectl get secrets -n openshift-pipelines signing-secrets -o jsonpath='{.data.cosign\.pub}' 2>/dev/null)
+done
+for NAMESPACE_SUFFIX in "development" "prod" "stage"; do
+    APP_NAMESPACE="${NAMESPACE}-app-${NAMESPACE_SUFFIX}"
+
+    cat <<EOF | kubectl apply -f - >/dev/null
+apiVersion: v1
+kind: Namespace
+metadata:
+  labels:
+    argocd.argoproj.io/managed-by: $NAMESPACE
+  name: $APP_NAMESPACE
+EOF
+
+    SECRET_NAME="cosign-pub"
+    if [ -n "$COSIGN_SIGNING_PUBLIC_KEY" ]; then
+        cat <<EOF | kubectl apply -f - >/dev/null
+apiVersion: v1
+data:
+    cosign.pub: $COSIGN_SIGNING_PUBLIC_KEY
+kind: Secret
+metadata:
+    labels:
+        app.kubernetes.io/instance: default
+        app.kubernetes.io/part-of: tekton-chains
+        operator.tekton.dev/operand-name: tektoncd-chains
+    name: $SECRET_NAME
+    namespace: $APP_NAMESPACE
+type: Opaque
+EOF
+        echo -n "."
+    fi
+    SECRET_NAME="gitlab-auth-secret"
+    if [ -n "$GITLAB__TOKEN" ]; then
+        kubectl -n $APP_NAMESPACE create secret generic "$SECRET_NAME" \
+            --from-literal=password=$GITLAB__TOKEN \
+            --from-literal=username=oauth2 \
+            --type=kubernetes.io/basic-auth \
+            --dry-run=client -o yaml | kubectl -n $APP_NAMESPACE apply --filename - --overwrite=true >/dev/null
+        echo -n "."
+    fi
+    SECRET_NAME="gitops-auth-secret"
+    if [ -n "$GITOPS__GIT_TOKEN" ]; then
+        kubectl -n $APP_NAMESPACE create secret generic "$SECRET_NAME" \
+            --from-literal=password=$GITOPS__GIT_TOKEN \
```

Comment:
We will need to also provide

```
--from-literal=username="oauth2" \
```

It is because if we create secret without username, the pipelines we are using on the main branch will be broken. See [comment](https://github.com/redhat-ai-dev/ai-lab-template/pull/36#discussion_r1767504108)

Reply:
See the gitlab-auth-secret above for another reference where we provided username

Reply:
In the current state of the pipeline definitions I have there is no `gitops-auth-secret` since I removed it when it was causing issues, so whatever we decide to do for this I can adapt the definitions too. That being said if we don't include it here _eventually_ the pipelines on the `main` branch won't be broken since the pipeline definition will change

Reply:
> In the current state of the pipeline definitions I have there is no `gitops-auth-secret` since I removed it when it was causing issues, so whatever we decide to do for this I can adapt the definitions too. That being said if we don't include it here _eventually_ the pipelines on the `main` branch won't be broken since the pipeline definition will change

@maysunfaisal @Jdubrick I wonder if this could be at least somewhat solved by adding an extra optional env var to provide a username and if its blank set up the classical way and if its not blank set up using the username flags. What do you all think?

Reply:
> > In the current state of the pipeline definitions I have there is no `gitops-auth-secret` since I removed it when it was causing issues, so whatever we decide to do for this I can adapt the definitions too. That being said if we don't include it here _eventually_ the pipelines on the `main` branch won't be broken since the pipeline definition will change
> 
> @maysunfaisal @Jdubrick I wonder if this could be at least somewhat solved by adding an extra optional env var to provide a username and if its blank set up the classical way and if its not blank set up using the username flags. What do you all think?

If there's more work to be done, we could look into separating into its own task to see what else is needed to get username based auth to work as well without removing classic compatibility. Only obstacle I still see with the toggle under the configure script is that once a deployment is configured either way it will limit it to just the software templates with whichever setup it supports.

Reply:
the username could just be hardcoded like it is in rhtap also @michael-valdron